lib/plugins/aws/provider.js

```diff
@@ -757,6 +768,37 @@ class AwsProvider {
             },
             iamManagedPolicies: { type: 'array', items: { $ref: '#/definitions/awsArnString' } },
             iamRoleStatements: { $ref: '#/definitions/awsIamPolicyStatements' },
+            images: {
+              type: 'object',
+              patternProperties: {
+                // TODO: VALIDATE THAT LENGTH
+                '^[a-z]{1,64}$': {
+                  oneOf: [
+                    // TODO: Combine into one?
```

Comment:
I think it's best to stick to one option, and validate exclusivity of `path` and `uri` manually.

It's because in next PR will introdoce common options that apply to all scenarios and we should not duplicate their definitions. Other reason is that it'll make validation errors more human 
friendly, `anyOf` fail in such case could be reported just with `provider.images.foo format is not supported` which doesn't inform what properties didn't validate

Otherwise (as a side note), let's always rely on `anyOf` (not `oneOf`)

_With `oneOf` AJV always validates all paths (if more that one passes validation fails), and applies coercion if needed in each case, which may malform final value._

_With `anyOf` AJV stops at first passing option_

Reply:
Thanks, that's a great advice, I'll change it